packages/react-components/react-table/src/components/DataGrid/DataGrid.types.ts

```diff
@@ -25,7 +26,10 @@ export type DataGridContextValue = HeadlessTableState<any> & {
  */
 export type DataGridProps = TableProps &
   Pick<DataGridContextValue, 'items' | 'columns'> &
-  Pick<Partial<DataGridContextValue>, 'focusMode'>;
+  Pick<Partial<DataGridContextValue>, 'focusMode'> &
+  Pick<UseSortOptions, 'sortState' | 'defaultSortState'> & {
+    onSortChange?: (e: React.MouseEvent, sortState: SortState) => void;
```

Comment:
```suggestion
    onSortChange?(e: React.MouseEvent, sortState: SortState): void;
```

This will solve your problems here https://github.com/microsoft/fluentui/pull/25655/files#r1023830620

Reply:
unfortunately changing this one will fail conformance tests since it can no longer find the correct type for the callback argument